File: app/src/main/java/com/mediaplayer/services/CleanUpService.java
package com.mediaplayer.services;


import android.content.Context;

import com.mediaplayer.variables.Config;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class CleanUpService {

    public static void deleteTempPlayBack(List<String> videoPath, Context ctx) {
        List<String> filesToDelete = new ArrayList<>();
        File internalFile[] = ctx.getApplicationContext().getFilesDir().listFiles();
        for (int i = 0; i < internalFile.length; i++) {
            String internalFileName = internalFile[i].getName();
            Boolean fileFound = false;
            for (int j = 0; j < videoPath.size(); j++) {
                String videoFile = new File(videoPath.get(j)).getName() + ".txt";
                if (internalFileName.equals(videoFile)) {
                    fileFound = true;
                    break;
                }
            }
            if (!fileFound) {
                filesToDelete.add(internalFileName);
            }
        }
        for (int k = 0; k < filesToDelete.size(); k++) {
            File file = new File(ctx.getApplicationContext().getFilesDir(), filesToDelete.get(k));
            boolean deleted = file.delete();
            if (!deleted) {
                System.out.println("Couldn't delete:::: " + filesToDelete.get(k));
            } else {
                System.out.println("deleted File:::: " + filesToDelete.get(k));
            }
        }
    }

    public static void deleteThumbnail(List<String> videoPath) {
        List<String> filesToDelete = new ArrayList<>();
        File imageFile[] = new File(Config.baseThumbPath + "/.dthumb").listFiles();
        if (imageFile==null){
            System.out.println("Found no list of items...");
            return;
        }
        for (int i = 0; i < imageFile.length; i++) {
            String thumbFileName = imageFile[i].getName();
            Boolean fileFound = false;
            for (int j = 0; j < videoPath.size(); j++) {
                String videoFile = new File(videoPath.get(j)).getName() + ".bmp";
                if (thumbFileName.equals(videoFile)) {
                    fileFound = true;
                    break;
                }
            }
            if (!fileFound) {
                filesToDelete.add(thumbFileName);
            }
        }
        for (int k = 0; k < filesToDelete.size(); k++) {
            File file = new File(Config.baseThumbPath + "/.dthumb", filesToDelete.get(k));
            boolean deleted = file.delete();
            if (!deleted) {
                System.out.println("Couldn't delete:::: " + filesToDelete.get(k));
            } else {
                System.out.println("deleted File:::: " + filesToDelete.get(k));
            }
        }
    }

}

File: app/src/main/java/com/mediaplayer/services/CleanUpService.kt
package com.mediaplayer.services


import android.content.Context

import com.mediaplayer.variables.Config

import java.io.File
import java.util.ArrayList

object CleanUpService {

    fun deleteTempPlayBack(videoPath: List<String>, ctx: Context) {
        val filesToDelete = ArrayList<String>()
        val internalFile = ctx.applicationContext.filesDir.listFiles()
        for (i in internalFile.indices) {
            val internalFileName = internalFile[i].name
            var fileFound: Boolean? = false
            for (j in videoPath.indices) {
                val videoFile = File(videoPath[j]).name + ".txt"
                if (internalFileName == videoFile) {
                    fileFound = true
                    break
                }
            }
            if ((!fileFound!!)) {
                filesToDelete.add(internalFileName)
            }
        }
        for (k in filesToDelete.indices) {
            val file = File(ctx.applicationContext.filesDir, filesToDelete[k])
            val deleted = file.delete()
            if (!deleted) {
                println("Couldn't delete:::: " + filesToDelete[k])
            } else {
                println("deleted File:::: " + filesToDelete[k])
            }
        }
    }

    fun deleteThumbnail(videoPath: List<String>) {
        val filesToDelete = ArrayList<String>()
        val imageFile = File(Config.baseThumbPath + "/.dthumb").listFiles()
        if (imageFile == null) {
            println("Found no list of items...")
            return
        }
        for (i in imageFile.indices) {
            val thumbFileName = imageFile[i].name
            var fileFound: Boolean? = false
            for (j in videoPath.indices) {
                val videoFile = File(videoPath[j]).name + ".bmp"
                if (thumbFileName == videoFile) {
                    fileFound = true
                    break
                }
            }
            if ((!fileFound!!)) {
                filesToDelete.add(thumbFileName)
            }
        }
        for (k in filesToDelete.indices) {
            val file = File(Config.baseThumbPath + "/.dthumb", filesToDelete[k])
            val deleted = file.delete()
            if (!deleted) {
                println("Couldn't delete:::: " + filesToDelete[k])
            } else {
                println("deleted File:::: " + filesToDelete[k])
            }
        }
    }

}
